Vizzuality/forest-atlas-landscape-cms
# Feature/user permissions

Permit user to execute operations over datasets, widgets and metadata depending on their role

## Testing instructions

If you already have a user with admin permissions on the API and FA, I recommend you to create another user with the same email like this: `your_user+sa@vizzuality.com`. You can change the permissions of this new user with your existing admin user, so you can see the new permissions when the user is a Publisher or a Site Admin.

### Datasets

**Site Admin**

1. On the list of datasets, you should be able to see all datasets with edit and remove buttons available
2. Access to the edition of a dataset which does not pertain to your current user and complete all steps
3. You should see the changes in the dataset
4. Remove the edited dataset and check that it does not appear in the list
5. Access to the edition of a dataset of the current user and complete all steps
6. You should see the changes in the dataset
7. Remove the edited dataset and check that it does not appear in the list

**Publisher**

1. On the list of datasets, you should be able to see all datasets. However, you can only see the edit button available on those datasets which pertains to the current user.
2. Access to the edition of a dataset and complete all steps
3. You should see the changes on the dataset
4. Remove the edited dataset and check that it does not appear in the list

### Widgets

**Site Admin**

1. On the list of widgets, you should be able to see all widgets with edit and remove buttons available
2. Access to the edition of a widget which does not pertain to your current user and complete all steps
3. You should see the changes in the widget
4. Remove the edited widget and check that it does not appear in the list
5. Access to the edition of a widget of the current user and complete all steps
6. You should see the changes in the widget
7. Remove the edited widget and check that it does not appear in the list

**Publisher**

1. On the list of widgets, you should be able to see all the widgets. However, you can only see the edit button available on those widgets which pertains to the current user.
2. Access to the edition of a widget and complete all steps
3. You should see the changes on the widget
4. Remove the edited widget and check that it does not appear in the list

## Pivotal Tracker

[https://www.pivotaltracker.com/story/show/170070837](https://www.pivotaltracker.com/story/show/170070837)
